tsdb/engine/tsm1/cache.go

```diff
@@ -247,7 +261,11 @@ func (c *Cache) merged(key string) Values {
 	var entries []*entry
 	sz := 0
 	for _, s := range c.snapshots {
-		e := s.store[key]
```

Comment:
This is the second race condition that the test case detects. Have added a read lock on the snapshot